### PR TITLE
Support enabling different triple colon extensions on demand

### PR DESCRIPTION
**Pull request by JialiYang-96**
QnA needs to use image triple colon extension only. So make extensions configurable.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/docfx?pullRequest=7715) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/docfx?pullRequest=7715) before we can accept your contribution.<br/><br/>:x: JialiYang-96 [sign now](https://cla.dotnetfoundation.org/dotnet/docfx?pullRequest=7715)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/docfx?pullRequest=7715) it.</sub>

File: src/Microsoft.Docs.MarkdigExtensions/MarkdownExtensions.cs [hunk 1]
```diff
@@ -139,9 +139,9 @@ public static MarkdownPipelineBuilder UseNestedColumn(this MarkdownPipelineBuild
             return pipeline;
         }
 
-        public static MarkdownPipelineBuilder UseTripleColon(this MarkdownPipelineBuilder pipeline, MarkdownContext context)
+        public static MarkdownPipelineBuilder UseTripleColon(this MarkdownPipelineBuilder pipeline, MarkdownContext context, TripleColonExtensionType enabledExtensionTypes = (TripleColonExtensionType)~0)
```
Comment (yufeih) [inline src/Microsoft.Docs.MarkdigExtensions/MarkdownExtensions.cs:142]: Changing this method signature breaks library contract and could potentially break libraries depending on it. Add a new method that can opt-in to triple colon extensions individually. We also don't need an enum.

```csharp
public static MarkdownPipelineBuilder UseTripleColon<T>(this MarkdownPipelineBuilder pipeline, MarkdownContext context, T extension) where T : ITripleColonExtensionInfo
```

File: src/Microsoft.Docs.MarkdigExtensions/TripleColon/TripleColonExtension.cs [hunk 2]
```diff
@@ -39,6 +39,14 @@ public TripleColonExtension(MarkdownContext context)
             }).ToDictionary(x => x.Name);
         }
 
+        public TripleColonExtension(MarkdownContext context, List<ITripleColonExtensionInfo> extension)
```
Comment (yufeih) [inline src/Microsoft.Docs.MarkdigExtensions/TripleColon/TripleColonExtension.cs:42]: forward the above constructor

File: src/Microsoft.Docs.MarkdigExtensions/MarkdownExtensions.cs [hunk 3]
```diff
@@ -145,6 +146,12 @@ public static MarkdownPipelineBuilder UseTripleColon(this MarkdownPipelineBuilde
             return pipeline;
         }
 
+        public static MarkdownPipelineBuilder UseTripleColon(this MarkdownPipelineBuilder pipeline, MarkdownContext context, List<ITripleColonExtensionInfo> extensions)
```
Comment (yufeih) [inline src/Microsoft.Docs.MarkdigExtensions/MarkdownExtensions.cs:149]:
```suggestion
        public static MarkdownPipelineBuilder UseTripleColon(this MarkdownPipelineBuilder pipeline, MarkdownContext context, params ITripleColonExtensionInfo[] extensions)
```

**Review by yufeih — APPROVED** [review]

File: src/Microsoft.Docs.MarkdigExtensions/TripleColon/TripleColonExtension.cs (hunk 2 above)
Comment (JialiYang-96) [reply src/Microsoft.Docs.MarkdigExtensions/TripleColon/TripleColonExtension.cs]: Revised

File: src/Microsoft.Docs.MarkdigExtensions/MarkdownExtensions.cs (hunk 3 above)
Comment (JialiYang-96) [reply src/Microsoft.Docs.MarkdigExtensions/MarkdownExtensions.cs]: Revised

**Pull request merged**